preupg/xmlgen/compose.py

```diff
@@ -77,8 +77,44 @@ def get_compose_dir_name(self):
 class ComposeXML(object):
 
     @staticmethod
-    def collect_group_xmls(source_dir, content=None, level=0,
+    def collect_group_xmls(module_set_dir, source_dir, content=None,
                            generate_from_ini=True):
+        """
+        Find group.xml file recursively through all module directories
+        and modules. Collect data from each of them into dictionary.
+
+        @param {str} module_set_dir - directory where all modules are stored
+        @param {str} source_dir - directory path for processing
+        @param {str} content - module result e.g. failed,needs_action,pass,...
+        @param {bool} generate_from_ini - True if xccdf-compose tool is used
+
+        @return {dict} - structure is file based, keys are top level module
+        directories, values are tuples which consist of 2 elements:
+            [0] - XML data for top level dir itself
+            [1] - dict of all modules inside [0] directory
+        @example
+        from file structure:
+            - RHEL6_7
+                - services (modules directory)
+                    - group.xml
+                    - group.ini
+                    - httpd (module)
+                        - group.xml
+                        - ...
+                    - tomcat (module)
+                        - group.xml
+                        - ...
+                - drivers (modules directory)
+                - ...
+        result is:
+            {services: tuple(<Element {services}>, {
+                httpd: tuple(<Element {httpd}>, {}),
+                tomcat: tuple(<Element {tomcat}>, {}),
+                ...
+            )},
```

Comment:
The indentation makes the result confusing.
Also this one line 114 has should be "}),".
I would make the dictionary more clearly indented:
```
            {
                services: tuple(
                    <Element {services}>,
                    {
                        httpd: tuple(<Element {httpd}>, {}),
                        tomcat: tuple(<Element {tomcat}>, {}),
                        ...
                    }
                ),
                drivers: tuple(
                    <Element {drivers}>,
                    {...}
                ),
                ...
            }
```